vms/platformvm/reward/calculator.go

```diff
@@ -10,53 +10,42 @@ import (
 	"github.com/ava-labs/avalanchego/utils/math"
 )
 
-var _ Calculator = (*calculator)(nil)
-
-type Calculator interface {
-	Calculate(stakedDuration time.Duration, stakedAmount, currentSupply uint64) uint64
-}
-
-type calculator struct {
-	maxSubMinConsumptionRate *big.Int
-	minConsumptionRate       *big.Int
-	mintingPeriod            *big.Int
-	supplyCap                uint64
-}
-
-func NewCalculator(c Config) Calculator {
-	return &calculator{
-		maxSubMinConsumptionRate: new(big.Int).SetUint64(c.MaxConsumptionRate - c.MinConsumptionRate),
-		minConsumptionRate:       new(big.Int).SetUint64(c.MinConsumptionRate),
-		mintingPeriod:            new(big.Int).SetUint64(uint64(c.MintingPeriod)),
-		supplyCap:                c.SupplyCap,
-	}
-}
-
-// Reward returns the amount of tokens to reward the staker with.
+// Calculate returns the amount of tokens to reward a staker with.
 //
 // RemainingSupply = SupplyCap - ExistingSupply
 // PortionOfExistingSupply = StakedAmount / ExistingSupply
 // PortionOfStakingDuration = StakingDuration / MaximumStakingDuration
 // MintingRate = MinMintingRate + MaxSubMinMintingRate * PortionOfStakingDuration
 // Reward = RemainingSupply * PortionOfExistingSupply * MintingRate * PortionOfStakingDuration
-func (c *calculator) Calculate(stakedDuration time.Duration, stakedAmount, currentSupply uint64) uint64 {
+func Calculate(
+	maxConsumptionRate uint64,
+	minConsumptionRate uint64,
+	mintingPeriod time.Duration,
+	supplyCap uint64,
```

Comment:
I am not oppsed to avoiding the interface/struct idea but I don't like this big list of parameters here.
There is a `rewards.Config` struct in this package that holds all of these configs. I'd change `Calculate` signature to be `Calculate(cfg Config, stakedDuration time.Duration, stakedAmount uint64, currentSupply uint64)`

Reply:
I thought about this in the original implementation but I thought it might be good to explicitly list these parameters because this might be used as a library function similar to `Split`.

I just wanted to avoid a world where `RewardConfig` gets something added which might make a downstream bug harder to find since it'll be initialized to zero to any code that forgets to update its config.

Reply:
I also thought to this Calculate method as another Split utility! And I still think it'd be ok to group the parameters in a reward.Config.
It's a bummer that golang does not provide a native way to mark a struct as const. However in platformvm we do pass around config.Config a lot and we do take care of never modifying it. We don't enforce it in the code (not sure there is a clean way to do it), but it a low risk situation.
A more relevant concern is the possible mis-initialization of config. I think when it comes to configs, we should initialize the in single place and just pass them around. Your change goes in this direction by avoiding building objects with a slice of that config (and there are UTs where the config was not duly initialized and you spot it).  I like your change exactly because it helped cleaning up the misconfigured test.